vms/platformvm/txs/builder/builder.go

```diff
@@ -4,524 +4,133 @@
-		nodeID ids.NodeID,
-		rewardAddress ids.ShortID,
-		keys []*secp256k1.PrivateKey,
-		changeAddr ids.ShortID,
-		memo []byte,
-	) (*txs.Tx, error)
-
-	// stakeAmount: amount the delegator stakes
-	// startTime: unix time they start delegating
-	// endTime: unix time they stop delegating
-	// nodeID: ID of the node we are delegating to
-	// rewardAddress: address to send reward to, if applicable
-	// keys: keys providing the staked tokens
-	// changeAddr: address to send change to, if there is any
-	NewAddPermissionlessDelegatorTx(
-		stakeAmount,
-		startTime,
-		endTime uint64,
-		nodeID ids.NodeID,
-		rewardAddress ids.ShortID,
-		keys []*secp256k1.PrivateKey,
-		changeAddr ids.ShortID,
-		memo []byte,
-	) (*txs.Tx, error)
-
-	// weight: sampling weight of the new validator
-	// startTime: unix time they start delegating
-	// endTime:  unix time they top delegating
-	// nodeID: ID of the node validating
-	// subnetID: ID of the subnet the validator will validate
-	// keys: keys to use for adding the validator
-	// changeAddr: address to send change to, if there is any
-	NewAddSubnetValidatorTx(
-		weight,
-		startTime,
-		endTime uint64,
-		nodeID ids.NodeID,
-		subnetID ids.ID,
-		keys []*secp256k1.PrivateKey,
-		changeAddr ids.ShortID,
-		memo []byte,
-	) (*txs.Tx, error)
-
-	// Creates a transaction that removes [nodeID]
-	// as a validator from [subnetID]
-	// keys: keys to use for removing the validator
-	// changeAddr: address to send change to, if there is any
-	NewRemoveSubnetValidatorTx(
-		nodeID ids.NodeID,
-		subnetID ids.ID,
-		keys []*secp256k1.PrivateKey,
-		changeAddr ids.ShortID,
-		memo []byte,
-	) (*txs.Tx, error)
-
-	// Creates a transaction that transfers ownership of [subnetID]
-	// threshold: [threshold] of [ownerAddrs] needed to manage this subnet
-	// ownerAddrs: control addresses for the new subnet
-	// keys: keys to use for modifying the subnet
-	// changeAddr: address to send change to, if there is any
-	NewTransferSubnetOwnershipTx(
-		subnetID ids.ID,
-		threshold uint32,
-		ownerAddrs []ids.ShortID,
-		keys []*secp256k1.PrivateKey,
-		changeAddr ids.ShortID,
-		memo []byte,
-	) (*txs.Tx, error)
-}
-
 func New(
 	ctx *snow.Context,
 	cfg *config.Config,
-	clk *mockable.Clock,
-	fx fx.Fx,
 	state state.State,
 	atomicUTXOManager avax.AtomicUTXOManager,
-	utxoSpender utxo.Spender,
-) Builder {
-	return &builder{
-		AtomicUTXOManager: atomicUTXOManager,
-		Spender:           utxoSpender,
-		state:             state,
-		cfg:               cfg,
-		ctx:               ctx,
-		clk:               clk,
-		fx:                fx,
+) *Builder {
+	return &Builder{
+		ctx:     ctx,
+		cfg:     cfg,
+		backend: NewBackend(set.Set[ids.ShortID]{}, state, atomicUTXOManager),
 	}
 }
 
-type builder struct {
-	avax.AtomicUTXOManager
-	utxo.Spender
-	state state.State
-
-	cfg *config.Config
-	ctx *snow.Context
-	clk *mockable.Clock
-	fx  fx.Fx
+type Builder struct {
+	ctx     *snow.Context
+	cfg     *config.Config
+	backend *Backend
```

Comment:
imo `backend` shouldn't be on the `Builder` struct. It should be created inside `builders()`. This will remove the weird hack we have around setting `b.backend.addrs` inside `builders()`.